tmt/utils.py

```diff
@@ -1449,6 +1449,59 @@ def option_to_key(option: str) -> str:
     return option.replace('-', '_')
 
 
+@dataclasses.dataclass
+class FieldMetadata(Generic[T]):
+    """
+    A dataclass metadata container used by our custom dataclass field management.
+
+    Attached to fields defined with :py:func:`field`
+    """
+
+    #: CLI option parameters, for lazy option creation.
+    option_args: Optional['FieldCLIOption'] = None
+    option_kwargs: Optional[Dict[str, Any]] = None
+    option_choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None
+
+    #: A :py:func:`click.option` decorator defining a corresponding CLI option.
+    _option: Optional['tmt.options.ClickOptionDecoratorType'] = None
+
+    #: A normalization callback to call when loading the value from key source
+    #: (performed by :py:class:`NormalizeKeysMixin`).
+    normalize_callback: Optional['NormalizeCallback[T]'] = None
+
+    @property
+    def option(self) -> Optional['tmt.options.ClickOptionDecoratorType']:
+        if self._option is None and self.option_args and self.option_kwargs:
+            if isinstance(self.option_choices, (list, tuple)):
+                self.option_kwargs['type'] = click.Choice(self.option_choices)
+
+            elif callable(self.option_choices):
+                self.option_kwargs['type'] = click.Choice(self.option_choices())
+
+            if self.normalize_callback is not None:
+                self.option_kwargs['type'] = self.normalize_callback
+
+            self._option = click.option(
+                *self.option_args,
+                **self.option_kwargs
+                )
+
+        return self._option
+
+
+def dataclass_field_by_name(cls: Any, name: str) -> 'dataclasses.Field[T]':
+    for field in dataclasses.fields(cls):
+        if field.name == name:
+            return field
+
+    else:
+        raise GeneralError(f"Could not find field '{name}' in class '{cls.__name__}'.")
+
+
+def dataclass_field_metadata(field: 'dataclasses.Field[T]') -> 'FieldMetadata[T]':
```

Comment:
Please, add a docstring.

Reply:
Fixed in e5f266bef1df4b011ed20040575f778189aa001f